python/ray/data/datasource/file_based_datasource.py

```diff
@@ -919,3 +937,38 @@ def _fetch_metadata_parallel(
         fetch_tasks.append(remote_fetch_func.remote(uri_chunk))
     results = metadata_fetch_bar.fetch_until_complete(fetch_tasks)
     yield from itertools.chain.from_iterable(results)
+
+
+def _open_file_with_retry(
+    file_path: str,
+    open_file: Callable[[], "pyarrow.NativeFile"],
+) -> "pyarrow.NativeFile":
+    """Open file with an exponential backoff retry strategy.
+
+    This is to avoid transient task failure with remote storage (such as S3),
+    when the remote storage throttles the requests.
+    """
+    import random
+    import time
+
+    for i in range(OPEN_FILE_RETRY_MAX_ATTEMPTS):
```

Comment:
Can we rewrite this to guard against OPEN_FILE_RETRY_MAX_ATTEMPTS set to 0? The name is also slightly confusing (can't tell if it's the total number of attempts or number of retries).

Reply:
Let me rename to `OPEN_FILE_MAX_ATTEMPTS`, and throw exception if it'set smaller than 1.

Reply:
Updated.